common/lib/xmodule/xmodule/contentstore/content.py

```diff
@@ -127,11 +134,100 @@ def get_location_from_path(path):
+
+        If a static asset CDN is configured, this path will include it.
+        Otherwise, the path will simply be relative.
+
+        Args:
+            course_key: key to the course which owns this asset
+            path: the path to said content
+
+        Returns:
+            string: fully-qualified path to asset
+        """
+
+        # Break down the input path.
+        _, base_url, relative_path, params, query_string, fragment = urlparse(path)
+
+        # Convert our path to an asset key.
+        asset_key = StaticContent.get_asset_key_from_path(course_key, relative_path)
+
+        # Check to see if we can actually provide a CDN-ified version
+        # of this asset back to the caller.
+        is_locked = False
+        try:
+            content = AssetManager.find(asset_key, as_stream=False)
+            is_locked = getattr(content, "locked", False)
+        except (ItemNotFoundError, NotFoundError):
+            # TODO: What's the right fallback here?  Assume it's locked?
+            # Can we ever reasonably except this state since we're actually
+            # calling the same store that will be asked to serve the asset?
+            pass
+
+        # Update any query parameter values that have asset paths in them.
+        query_params = parse_qsl(query_string)
+        updated_query_params = []
+        for query_name, query_value in query_params:
+            if query_value.startswith("/static/"):
+                offset = len('/static/')
+                new_query_value = unicode(StaticContent.get_asset_key_from_path(course_key, query_value[offset:]))
+                if not new_query_value.startswith('/'):
+                    new_query_value = '/' + new_query_value
+                updated_query_params.append((query_name, new_query_value))
+            else:
+                updated_query_params.append((query_name, query_value))
+
+        # Figure out if we should put in the base URL.
+        base_url = None
+        if not is_locked:
+            base_url = StaticContent.base_url
+
+        prefixed_asset_key = unicode(asset_key)
+        if not prefixed_asset_key.startswith('/'):
+            prefixed_asset_key = '/' + prefixed_asset_key
+
+        return urlunparse((None, base_url, prefixed_asset_key, params, urlencode(updated_query_params), fragment))
+
+    @staticmethod
+    def get_asset_key_from_path(course_key, path):
+        """
+        Parses a path, extracting an asset key or creating one.
+
+        Args:
+            course_key: key to the course which owns this asset
+            path: the path to said content
+
+        Returns:
+            AssetKey: the asset key that represents the path
+        """
+
+        # If the path starts with /static/, just drop it.
+        if path.startswith('/static/'):
+            path = path[len('/static/'):]
+
+        # Skip leading forward slash if present.
+        start_position = 1 if path.startswith('/') else 0
```

Comment:
Maybe a helper method that will remove a prefix if it exists and add a leading '/' if it's necessary?

```python
path = remove_prefix(path, '/static/')
```